packages/impit-client/src/index.ts

```diff
@@ -143,44 +155,30 @@ export class ImpitHttpClient implements BaseHttpClient {
         return {
             headers: response.headers,
             statusCode: response.status,
-            url,
+            url: typeof request.url === 'string' ? request.url : request.url.href,
             request,
-            redirectUrls: redirects?.redirectUrls ?? [],
+            redirectUrls,
             trailers: {},
             body: responseBody,
             complete: true,
         };
     }
 
-    /**
-     * @inheritDoc
-     */
-    async sendRequest<TResponseType extends keyof ResponseTypes>(
-        request: HttpRequest<TResponseType>,
-    ): Promise<HttpResponse<TResponseType>> {
-        return this.performRequest(request);
-    }
-
     /**
      * @inheritDoc
      */
     async stream(request: HttpRequest): Promise<StreamingHttpResponse> {
-        const response = await this.performRequest(request);
-
-        const stream = new Readable();
-        stream.push(response.body);
-        stream.push(null);
+        const [response, redirectUrls] = await this.getResponse(request);
 
         return {
             request,
-            url: response.url,
-            ip: response.ipAddress,
-            statusCode: response.statusCode,
-            stream,
+            url: redirectUrls[redirectUrls.length - 1].href,
+            statusCode: response.status,
+            stream: Readable.fromWeb(response.body as ReadableStream<any>),
             complete: true,
-            downloadProgress: { percent: 100, transferred: response.body.length },
```

Comment:
I cannot imagine the correct implementation of this (and / or `uploadProgress` below). I guess we could read `headers.content-length` to get the "100%" info, but to see the amount of "transferred" data, we'd need to tap into the stream, which would cause us to consume it before the user does.

Do you have any idea here @janbuchar ?

Reply:
(Plus, right here it's a POJO, repeated reads wouldn't give any different info, if we don't use a getter. Either way, the API feels quite unintuitive - I'm assuming it's legacy?)

Reply:
Yes, this is modelled after `got` for BC. And you're right that the correct implementation would be an object with a getter that returns some internal counter of received chunks vs total chunks determined from Content-Length.